CMakeLists.txt

```diff
@@ -91,6 +91,7 @@ if (NOT WIN32)
     set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 0unstable)
 
     target_compile_definitions(${PROJECT_NAME} PUBLIC)
+    target_compile_options(${PROJECT_NAME} PRIVATE "-Wall" "-Werror" "-Wpedantic")
```

Comment:
would be better to have it in a separate commit

Reply:
Only for the SDK library, or for the library and the binaries?

Reply:
Re-pushed the branch, making this line a separate commit.

Reply:
meant for both, but was more a nit